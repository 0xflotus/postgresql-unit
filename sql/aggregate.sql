-- test aggregates

CREATE TEMP TABLE u (
	u unit
);

SELECT sum(u) AS null_sum FROM u;
SELECT min(u) AS null_min FROM u;
SELECT max(u) AS null_max FROM u;

INSERT INTO u VALUES (meter()), (meter(2));
SELECT sum(u) AS three_meters FROM u;
SELECT min(u) AS one_meter FROM u;
SELECT max(u) AS two_meters FROM u;

DELETE FROM u;
INSERT INTO u VALUES (kilogram(2)), (kilogram());
SELECT sum(u) AS three_kilogram FROM u;
SELECT min(u) AS one_kilogram FROM u;
SELECT max(u) AS two_kilogram FROM u;

DELETE FROM u;
INSERT INTO u VALUES (ampere()), (kilogram(2));
SELECT sum(u) AS error FROM u;
SELECT min(u) AS error FROM u;
SELECT max(u) AS error FROM u;
