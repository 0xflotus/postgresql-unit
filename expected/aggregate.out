-- test aggregates
CREATE TEMP TABLE u (
	u unit
);
SELECT sum(u) AS null_sum FROM u;
 null_sum 
----------
 
(1 row)

SELECT min(u) AS null_min FROM u;
 null_min 
----------
 
(1 row)

SELECT max(u) AS null_max FROM u;
 null_max 
----------
 
(1 row)

SELECT avg(u) AS null_avg FROM u;
 null_avg 
----------
 
(1 row)

INSERT INTO u VALUES (meter()), (meter(2));
SELECT sum(u) AS three_meters FROM u;
 three_meters 
--------------
 3 m
(1 row)

SELECT min(u) AS one_meter FROM u;
 one_meter 
-----------
 1 m
(1 row)

SELECT max(u) AS two_meters FROM u;
 two_meters 
------------
 2 m
(1 row)

SELECT avg(u) AS oneandhalf_meters FROM u;
 oneandhalf_meters 
-------------------
 1.5 m
(1 row)

DELETE FROM u;
INSERT INTO u VALUES (kilogram(2)), (kilogram());
SELECT sum(u) AS three_kilogram FROM u;
 three_kilogram 
----------------
 3 kg
(1 row)

SELECT min(u) AS one_kilogram FROM u;
 one_kilogram 
--------------
 1 kg
(1 row)

SELECT max(u) AS two_kilogram FROM u;
 two_kilogram 
--------------
 2 kg
(1 row)

SELECT avg(u) AS oneandhalf_kilogram FROM u;
 oneandhalf_kilogram 
---------------------
 1.5 kg
(1 row)

INSERT INTO u VALUES (NULL);
SELECT sum(u) AS three_kilogram FROM u;
 three_kilogram 
----------------
 3 kg
(1 row)

SELECT min(u) AS one_kilogram FROM u;
 one_kilogram 
--------------
 1 kg
(1 row)

SELECT max(u) AS two_kilogram FROM u;
 two_kilogram 
--------------
 2 kg
(1 row)

SELECT avg(u) AS oneandhalf_kilogram FROM u;
 oneandhalf_kilogram 
---------------------
 1.5 kg
(1 row)

DELETE FROM u;
INSERT INTO u VALUES (ampere()), (kilogram(2));
SELECT sum(u) AS error FROM u;
ERROR:  units are not the same: "1 A", "2 kg"
SELECT min(u) AS error FROM u;
ERROR:  units are not the same: "1 A", "2 kg"
SELECT max(u) AS error FROM u;
ERROR:  units are not the same: "1 A", "2 kg"
SELECT avg(u) AS error FROM u;
ERROR:  units are not the same: "1 A", "2 kg"
CONTEXT:  SQL function "unit_accum" statement 1
